src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs

```diff
@@ -140,37 +151,23 @@ public static void ReleaseJSOwnedObjectByGCHandle(JSMarshalerArgument* arguments
         {
             ref JSMarshalerArgument arg_exc = ref arguments_buffer[0]; // initialized by caller in alloc_stack_frame()
             ref JSMarshalerArgument arg_1 = ref arguments_buffer[2]; // initialized and set by caller
+
             try
             {
-                var gcHandle = arg_1.slot.GCHandle;
-                if (IsGCVHandle(gcHandle))
-                {
-                    if (ThreadJsOwnedHolders.Remove(gcHandle, out PromiseHolder? holder))
-                    {
-                        holder.GCHandle = IntPtr.Zero;
-                        holder.Callback!(null);
-                    }
-                }
-                else
-                {
-                    GCHandle handle = (GCHandle)gcHandle;
-                    var target = handle.Target!;
-                    if (target is PromiseHolder holder)
-                    {
-                        holder.GCHandle = IntPtr.Zero;
-                        holder.Callback!(null);
-                    }
-                    else
-                    {
-                        ThreadJsOwnedObjects.Remove(target);
-                    }
-                    handle.Free();
-                }
+                // when we arrive here, we assume that all proxies are owned by calling thread
+                var ctx = JSProxyContext.CapturedInstance = JSProxyContext.AssertCurrentContext();
```

Comment:
`CapturedInstance` is a weird name for this. If I understand it right, it's something like 'thread-local context for the current operation', it might be worth giving it a name that conveys some of that information more clearly. I don't have any particularly good ideas for it, but something like `CurrentOperationContext` could be better.

Reply:
I like CurrentOperationContext